flutter/lib/src/sentry_asset_bundle.dart

```diff
@@ -0,0 +1,97 @@
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+import 'package:flutter/material.dart';
+import 'package:sentry/sentry.dart';
+
+typedef _Parser<T> = Future<T> Function(String value);
+
+/// An [AssetBundle] which creates automatic performance traces for loading
+/// assets.
+///
+/// You can wrap other [AssetBundle]s in it:
+/// ```dart
+/// SentryAssetBundle(bundle: someOtherAssetBundle)
+/// ```
+/// If you're not providing any [AssetBundle], it falls back to the [rootBundle].
+///
+/// If you want to use the [SentryAssetBundle] by default you can achieve this
+/// with the following code:
+/// ```dart
+/// DefaultAssetBundle(
+///   bundle: SentryAssetBundle(),
+///   child: MaterialApp(
+///     home: MyScaffold(),
+///   ),
+/// );
+/// ```
+/// [Image.asset], for example, will then use [SentryAssetBundle].
+class SentryAssetBundle implements AssetBundle {
+  SentryAssetBundle({Hub? hub, AssetBundle? bundle})
+      : _hub = hub ?? HubAdapter(),
+        _bundle = bundle ?? rootBundle;
+
+  final Hub _hub;
+  final AssetBundle _bundle;
+
+  @override
+  Future<ByteData> load(String key) async {
+    final span = _hub.getSpan()?.startChild(
+          'file.read',
+          description: 'AssetBundle.load(key=$key)',
```

Comment:
let's align the description with the `file.read` from Java, https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/instrumentation/file/SentryFileInputStream.java
@romtsn can give you more details like extra fields as well when available e.g. file name and size

Reply:
yep, described it [here](https://github.com/getsentry/develop/issues/463#issuecomment-990042948)

Reply:
The `key` doesn't need to guarded by `sendDefaultPii` because it's a path relative to the pubspec.yaml. The other points we're changed accordingly.

Reply:
Left a comment about the `description` -> https://github.com/getsentry/sentry-dart/pull/685#discussion_r786228015